src/topology/order/priestley.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Sara Rousta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Sara Rousta
+-/
+import order.upper_lower
+import topology.separation
+
+/-!
+# Priestley spaces
+
+This file defines Priestley spaces. A Priestley space is an ordered compact topological space such
+that any two distinct points can be separated by a clopen upper set.
+
+## Main declarations
+
+* `priestley_space`: Prop-valued mixin stating the Priestley separation axiom: Any two distinct
+  points can be separated by a clopen upper set.
+
+## Implementation notes
+
+We do not include compactness in the definition, so a Priestley space is to be declared as follows:
+`[preorder α] [topological_space α] [compact_space α] [priestley_space α]`
```

Comment:
How about the following setup:
 * a mixin class `priestley_separated` containing the Priestley separation axiom (i.e. rename the current `priestley_space` class)
 * a class `priestley_space` extending both `priestley_separated` and `compact_space`
Then you still have access to the Priestley separation axiom as a mixin, while keeping the standard mathematical names intact.

A drawback is that you have to manually upgrade a Priestley separated space with a separate proof of compactness. The new instance synthesis in Lean 4 should make that automatic though, and these kinds of manual upgrades are already very common throughout mathlib (compare `inhabited` + `subsingleton` → `unique`).

On the other hand, the topology library seems to use a lot of mixins so the current approach might just be the accepted style.

Reply:
I am happy with both, but I don't believe the mixin extension to be that useful.